index.bs

```diff
@@ -52,72 +49,205 @@ RTCPeerConnection, canvas or MediaRecorder.
+    constructor(MediaStreamTrackGeneratorInit init)
+    attribute WritableStream writable;  // VideoFrame or AudioFrame
+    attribute ReadableStream readableControl;  // MediaStreamGeneratorSignal
 };
 
+dictionary MediaStreamTrackGeneratorInit {
+  // At least one of the two fields must be set.
+  // If both are provided and signalTarget.kind and kind do not match, the
+  // MediaStreamTrackGenerator's constructor will raise an exception.
+  MediaStreamTrack signalTarget;
+  MediaStreamTrackGeneratorKind kind;
+}
 </pre>
 
+### Internal slots
+<dl>
+<dt><dfn attribute for=MediaStreamTrackGenerator>[[signalTarget]]</dfn></dt>
+<dd>(Optional) track to which the {{MediaStreamTrackGenerator}} will automatically forward control signals.</dd>
+<dt><dfn attribute for=MediaStreamTrackGenerator>[[kind]]</dfn></dt>
+<dd>Kind of media data for this {{MediaStreamTrackGenerator}}. Must be a valid {{MediaStreamTrackKind}}
+  (i.e., `"audio"`  or `"video"`).</dd>
+</dl>
+
+### Constructor
+<dfn constructor for=MediaStreamTrackGenerator title="MediaStreamTrackGenerator(init)">
+  MediaStreamTrackGenerator(init)
+</dfn>
+1. If {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}} is not empty and is not a valid {{MediaStreamTrack}},
+   or {{init.}}{{MediaStreamTrackGenerator/[kind]}} is not empty and it is not `"audio"` or `"video"`,
+   throw a {{TypeError}}.
+2. If neither {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}} nor
+   and {{init.}}{{MediaStreamTrackGenerator/[kind]}} are empty, and
+   {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}}{{.}}{{MediaStreamTrack/kind}}
+   does not match {{init.}}{{MediaStreamTrackGeneratorInit/kind}}, throw a {{TypeError}}.
+3. Let |g| be a new {{MediaStreamTrackGenerator}} object.
+4. If {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}} is not empty,
+   assign {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}}
+   to the {{MediaStreamTrackGenerator/[signalTarget]}} internal slot of |g| and
+   initialize the {{MediaStreamTrack/kind}} field of |g| (inherited from {{MediaStreamTrack}})
+   with {{init.}}{{MediaStreamTrackGeneratorInit/signalTarget}}{{.}}{{MediaStreamTrack/kind}}.
+   Otherwise, initialize the {{MediaStreamTrack/kind}} field of |g| with
+   {{init.}}{{MediaStreamTrackGenerator/[kind]}}.
+5. Return |g|.
+
+### Attributes
+<dl>
+<dt><dfn attribute for=MediaStreamTrackGenerator>writable</dfn></dt>
+<dd>Allows writing media frames the {{MediaStreamTrackGenerator}}, which is
+itself a {{MediaStreamTrack}}. If the {{MediaStreamTrackGenerator/[kind]}} attribute is `"audio"`,
+the stream will accept {{AudioFrame}} objects and fail with any other type. If
+{{MediaStreamTrackGenerator/[kind]}} is `"video"`, the stream will accept {{VideoFrame}} objects
+and fail with any other type. When a frame is written to {{MediaStreamTrackGenerator/writable}},
+its `close()` method is automatically invoked, so that its internal resources are no longer
```

Comment:
should this be "destroyed"?
Also, "its" may be ambiguous, so say "the frame's"

Reply:
Done "the frame's".
Since the destroy() method was replaced by close(), I preferred to keep the reference to invoking the close() method, as it looks to me like the most accurate way to describe what happens.

Reply:
OK, the frame spec changed. Then it's OK.